src/components/Teams/Teams.jsx

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from "react";
+import TeamMember from "../TeamMember/TeamMember";
+
+const Teams = () => {
+  const [isLoadedTeamsData, setIsLoadedTeamsData] = useState(false);
+  const [teamsData, setTeamsData] = useState([]);
+  const [teamMembersData, setTeamMembersData] = useState([]);
+  const [teamsError, setTeamsError] = useState(null);
+
+  useEffect(() => {
+  
+    //TODO: Add abort controller when fetching from API
+
+    setIsLoadedTeamsData(false);
+    // Fetch data from the JSON file in the public directory
+    async function fetchData () {
+      try {
+        const [responseTeams, responseMembers] =  await Promise.all([fetch("/teams.json"), fetch("/members.json")]);
+        // Check if both responses are successful
+        if (!responseTeams.ok || !responseMembers.ok) {
+          setTeamsError('Failed to fetch data!');
+        };
+        const dataTeams = await responseTeams.json();
+        const dataMembers = await responseMembers.json();
+
+        setTeamsData(dataTeams);
+        setTeamMembersData(dataMembers);
+      } catch (error) {
+        setTeamsError(error);
+        console.error("Error fetching data:", error);
+      } finally {
+        setIsLoadedTeamsData(true);
+      }
+    }
+    fetchData();
+    
+    //TODO: Add cleanup function for abort controller when fetching from API
+    
+  }, []);
+
+  // If data not yet loaded
+  if (!isLoadedTeamsData) {
+    return <div>Loading...</div>;
+  }
+
+  // If error encountered
+  if (teamsError) {
+    return <div>An error occurred!</div>;
```

Comment:
```suggestion
    return <div>Unable to load teams. Please try refreshing the page.</div>;
```

As a UX improvement we give a more detailed error message, explaining what happened and also provide a potential fix for it.

[This heuristic](https://www.nngroup.com/articles/ten-usability-heuristics/#toc-9-help-users-recognize-diagnose-and-recover-from-errors-9) for UX design in specific should be present here.